validator/delphi_validator/validate.py

```diff
@@ -1,237 +1,721 @@
+# -*- coding: utf-8 -*-
+"""
+Tools to validate CSV source data, including various check methods.
+"""
+
 import sys
 import re
-import pandas as pd
-import numpy as np
-from pathlib import Path
-from itertools import product
-from datetime import date, datetime, timedelta
-from .datafetcher import *
 import math
+from os.path import join
+from datetime import date, datetime, timedelta
+import pandas as pd
 
-negated_regex_dict = {
-    'county': '^(?!\d{5}).*$',
-    'hrr': '^(?!\d{1,3}).*$',
-    'msa': '^(?!\d{5}).*$',
-    'state': '^(?![A-Z]{2}).*$',
-    'national': '(?!usa).*$'
+from .errors import ValidationError
+from .datafetcher import filename_regex, \
+    read_filenames, load_csv, get_geo_sig_cmbo, \
+    read_geo_sig_cmbo_files, fetch_api_reference
+
+# Recognized geo types.
+geo_regex_dict = {
+    'county': '^\d{5}$',
+    'hrr': '^\d{1,3}$',
+    'msa': '^\d{5}$',
+    'state': '^[a-z]{2}$',
+    'national': '^usa$'
```

Comment:
`national` requires a two-character country code

```suggestion
    'national': '^[a-z]{2}$'
```